group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -0,0 +1,868 @@
+                    topicsImage
+                );
+
+                if (!subscriptionMetadata.equals(group.subscriptionMetadata())) {
+                    log.info("[GroupId " + groupId + "] Computed new subscription metadata: "
+                        + subscriptionMetadata + ".");
+                    records.add(newGroupSubscriptionMetadataRecord(groupId, subscriptionMetadata));
+                }
+
+                groupEpoch += 1;
+                records.add(newGroupEpochRecord(groupId, groupEpoch));
+
+                log.info("[GroupId " + groupId + "] Bumped group epoch to " + groupEpoch + ".");
+            }
+
+            member = updatedMember;
+        }
+
+        // Update target assignment if needed. If the new target has any changes, we write the
+        // changes to the log.
+        int targetAssignmentEpoch = group.assignmentEpoch();
+        Assignment targetAssignment = group.targetAssignment(memberId);
+        if (groupEpoch > targetAssignmentEpoch) {
+            String preferredServerAssignor = group.preferredServerAssignor(
+                member.memberId(),
+                member.serverAssignorName()
+            ).orElse(defaultAssignor.name());
+
+            try {
+                TargetAssignmentBuilder.TargetAssignmentResult assignmentResult =
+                    new TargetAssignmentBuilder(groupId, groupEpoch, assignors.get(preferredServerAssignor))
+                        .withMembers(group.members())
+                        .withSubscriptionMetadata(subscriptionMetadata)
+                        .withTargetAssignment(group.targetAssignments())
+                        .addOrUpdateMember(member.memberId(), member)
+                        .build();
+
+                log.info("[GroupId " + groupId + "] Computed a new target assignment for epoch " + groupEpoch + ": "
+                    + assignmentResult.targetAssignment() + ".");
+
+                records.addAll(assignmentResult.records());
+                targetAssignment = assignmentResult.targetAssignment().get(member.memberId());
+                targetAssignmentEpoch = groupEpoch;
+            } catch (PartitionAssignorException ex) {
+                String msg = "Failed to compute a new target assignment for epoch " + groupEpoch + ": " + ex + ".";
+                log.error("[GroupId " + groupId + "] " + msg);
+                throw new UnknownServerException(msg, ex);
+            }
+        }
+
+        // If the member is stable and its next epoch matches the current target epoch
+        // of the assignment, we don't have to update its current assignment. Otherwise,
+        // we reconcile its current state based on the target assignment.
+        boolean assignmentUpdated = false;
+        if (member.state() != ConsumerGroupMember.MemberState.STABLE
+            || member.nextMemberEpoch() != targetAssignmentEpoch) {
+            updatedMember = new CurrentAssignmentBuilder(member)
```

Comment:
nit: can we add a newline between the if statement and updatedMember?

Reply:
Let me put the condition on one line.